### PR TITLE
Back-port support of newer pydot to make travis work with this 0.8.X …

### PR DESCRIPTION
**Pull request by nouiz**
…branch

This time I make a PR to make sure travis is fixed instead of doing it on the branch itself.

### CODE REVIEW

**Comment by nouiz** [comment]
jenkins failure can be ignored. We didn't make it work with the 0.8.X branch. I'm just tring to make travis work again with that branch.

File: theano/printing.py [hunk 1]
```diff
@@ -23,19 +23,33 @@
 from theano.compile.profilemode import ProfileMode
 
 pydot_imported = False
+pydot_imported_msg = ""
 try:
     # pydot-ng is a fork of pydot that is better maintained
     import pydot_ng as pd
     if pd.find_graphviz():
```
Comment (lamblin) [inline theano/printing.py]: I think the error is raised here:
```
AttributeError: 'module' object has no attribute 'find_graphviz'
```

File: .travis.yml [hunk 2]
```diff
@@ -38,7 +38,7 @@ install:
   - source activate pyenv
   - if [[ $TRAVIS_PYTHON_VERSION == '2.7' ]]; then pip install pydot; else pip install pydot-ng; fi
   - pip install . --no-deps
```
Comment (lamblin) [inline .travis.yml]: The log says:
```
$ pip install . --no-deps

Processing /home/travis/build/Theano/Theano

  Requirement already satisfied (use --upgrade to upgrade): Theano==0.8.2 from file:///home/travis/build/Theano/Theano in /home/travis/miniconda2/envs/pyenv/lib/python2.7/site-packages
```

So I'll try with `--upgrade`

Comment (lamblin) [inline .travis.yml]: Yep, this seems to actually be an issue with the dev branch as well...

File: theano/printing.py (hunk 1 above)
Comment (lamblin) [inline theano/printing.py]: The issue was that it was still using an older version of Theano.

File: theano/printing.py [hunk 3]
```diff
@@ -23,19 +23,33 @@
...
         pydot_imported = True
-except ImportError:
+    else:
+        pydot_imported_msg = "pydot-ng can't find graphviz. Install graphviz."
+except (ImportError, AttributeError):
```
Comment (lamblin) [inline theano/printing.py]: I don't think that's right, AttributeError should not be raised by pydot-ng, I mis-diagnosed that.

**Comment by lamblin** [comment]
Continued in #5360

**Pull request closed without merging**